sources/api/pluto/src/proxy.rs

```diff
@@ -90,11 +91,19 @@ where
         ));
     }
 
-    let https_connector = HttpsConnectorBuilder::new()
-        .with_native_roots()
+    let mut base_connector = HttpConnector::new_with_resolver(GaiResolver::new());
+    base_connector.enforce_http(false);
```

Comment:
I think it would be cleaner to put the hyper proxy dependency into the smithy client, rather than duplicating this code or exposing more of its methods:
https://gist.github.com/bcressey/d52538b4f7cb13dcbbdf984a0c0f713e

Apart from that, my only misgiving is where to put the `proxy::setup_proxy` interface so that both `pluto` and `cfsignal` can access it.

Reply:
> Apart from that, my only misgiving is where to put the proxy::setup_proxy interface so that both pluto and cfsignal can access it.

Maybe add this as a separate module to the smithy client fork for now, since it has the right dependencies already.